packages/graphql/src/translate/create-projection-and-params.ts

```diff
@@ -598,6 +598,23 @@ function createProjectionAndParams({
         return res;
     }
 
+    const existingProjection = { ...resolveTree.fieldsByTypeName[node.name] };
+
+    // if we have a global node, and the selection set includes the id field
+    if (node.isGlobalNode() && existingProjection.id) {
+        const globalIdField = node.getGlobalIdField();
+        // check to see if the idField has been projected, and include it if it hasn't
+        if (!Object.values(existingProjection).find((field) => field.name === globalIdField)) {
+            existingProjection[globalIdField] = {
+                alias: node.getGlobalIdField(),
+                args: {},
+                fieldsByTypeName: {},
+                name: node.getGlobalIdField(),
+            };
+        }
+        delete existingProjection.id;
+    }
```

Comment:
I could do with an explanation of this logic surrounding the projection of the `id` field!

Reply:
If a user includes the field "id" in the selectionSet for the node, then we need to make sure we include the underlying unique field in the projection for the `toGlobalId()` function.  E.g., with a type "Movie" that uses "_id" as it's unique field, if I query just for the "id" and "title", the field resolver for the ID will receive an object { "title": "something" }. That object needs to include the "_id" value.

Reply:
There is some interplay with [this](https://github.com/neo4j/graphql/pull/866#discussion_r832088478) issue above -- if you are right (and I think you are) and that "id" in the db can be the underlying field, then we'll need a conditional check before deleting the existingProject.id field. 

In general, if the underlying field is **not** "id" but rather "_id", "dbId", etc., then we have to remove the "id" field otherwise the db will err when it tries to return the null value.

Reply:
Yes, I think that conditional is needed - perfectly plausible that `id` will be the name of the property in the database.